app/routes/goal.py

```
@@ -0,0 +1,110 @@
+from flask import Blueprint, jsonify, request, make_response, abort
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+import requests
+from datetime import datetime
+import os
+from app.routes.task import validate_item
+
+goal_bp = Blueprint("goal", __name__, url_prefix="/goals")
+@goal_bp.route("", methods=["POST"])
+def create_one_goal():
+    request_body = request.get_json()
+    if not "title" in request_body:
+        return make_response({"details": "Invalid data"}, 400)
+    new_goal  = Goal.from_dict(request_body)
+    
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return make_response({"goal": new_goal.to_dict()}, 201)
+@goal_bp.route("", methods=["GET"])
+def get_goals():
+    response = []
+    title_query = request.args.get("title")
+    if title_query is None:
+        all_goals = Goal.query.all()
+    else:
+        all_goals = Goal.query.filter_by(title = title_query)
+    for goal in all_goals:
+        response.append(goal.to_dict())   
+    return jsonify(response), 200
+# @goal_bp.route("", methods=["GET"])
+# def get_goals():
+#     sort_direction = request.args.get("sort", default="desc")
+#     if sort_direction == "asc":
+#         all_goals = Goal.query.order_by(Goal.title.asc())
+#     else:
+#         all_goals = Goal.query.order_by(Goal.title.desc())
+    
+#     response = [goal.to_dict() for goal in all_goals]
+#     return jsonify(response), 200
+
+
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    goal = validate_item(Goal, goal_id)
+    return make_response({"goal": goal.to_dict()}, 200)
+
+@goal_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = validate_item(Goal, goal_id)
+    
+    request_data = request.get_json()
+
+    goal.title = request_data["title"]
```

I'd recommend nesting this in a try-except so that you can return an error message if the user does not supply a title in the request body.